File: Razmik/Week1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main()
{
    long cnum;
    char partly[16];
    bool luhn = false;
    int j = 0;
    string ctype = "INVALID";
    cnum = get_long("Number: ");
    int nDig = floor(log10(cnum)) + 1;
    if (nDig != 13 && nDig != 15 && nDig != 16)
    {
        printf("%s\n", ctype);
        return 0;
    }
    sprintf(partly, "%li", cnum);
    for (int i = 0; i < nDig ; i++)
        if ((i / 2) * 2 == i)
            if (partly[i] - '0' > 4)
            {
                j = j + (partly[i] - '5') * 2 + 1;
            }
            else
            {
                j = j + (partly[i] - '0') * 2;
            }
        else
        {
            j += partly[i] - '0';
        }
    if ((j / 10) * 10 == j)
    {
        luhn = true;
    }
    if (partly[0] == '4' && luhn)
    {
        ctype = "VISA";
    }
    else if (partly[0] == '3' && (partly[1] == '4' || partly[1] == '7'))
    {
        ctype = "AMEX";
    }
    else if (luhn && partly[0] == '5' && (partly[1] - '0' > 0 && partly[1] - '0' < 6))
    {
        ctype = "MASTERCARD";
    }
    printf("%s\n", ctype);
}

File: Razmik/Week6/credit.py
#!/usr/bin/python3.7
n = ""
p = 0
while type(n) is str:
    n = input("Number: ")
    if n.isnumeric():
        n = [int(i) for i in n]
if 13 < len(n) < 16 and len(n) != 14:
    for i in range(len(n)):
        p += n[::-1][i] if not i % 2 else (n[::-1][i] * 2) // 10 + (n[::-1][i] * 2 - ((n[::-1][i] * 2) // 10) * 10)
    if not p % 10 and n[0] == 4:
        exit("VISA\n")
    elif not p % 10 and n[0] == 5 and 1 <= n[1] <= 5:
        exit("MASTERCARD\n")
    elif (n[0] == 3 and (n[1] == 4 or n[1] == 7)):
        exit("AMEX\n")
print("INVALID\n")
